brightener.h

```diff
@@ -9,9 +12,9 @@ struct Image {
 
 class ImageBrightener {
 private:
-	Image m_inputImage;
+	unique_ptr<Image> m_inputImage;
 public:
-	ImageBrightener(Image& inputImage);
+	ImageBrightener(unique_ptr<Image>& inputImage);
 	int BrightenWholeImage();
-	Image GetImage();
+	unique_ptr<Image>& GetImage();
```

Comment:
Avoid applying reference type (`&`) to `unique_ptr`. The meaning of a `unique_ptr` is that there is only one pointer to the object at any time. Making a reference to it will go against that. When the `unique_ptr` goes out of scope, the object will be destroyed, while the reference will still be alive.

Reply:
removed the reference.
thanks i could understand the unique_ptr purpose